bson/bsonrw/extjson_reader.go

```diff
@@ -159,26 +159,18 @@ func (ejvr *extJSONValueReader) pop() {
 	}
 }
 
-func (ejvr *extJSONValueReader) skipDocument() error {
-	// read entire document until ErrEOD (using readKey and readValue)
-	_, typ, err := ejvr.p.readKey()
-	for err == nil {
-		_, err = ejvr.p.readValue(typ)
-		if err != nil {
-			break
-		}
-
-		_, typ, err = ejvr.p.readKey()
-	}
-
-	return err
-}
-
-func (ejvr *extJSONValueReader) skipArray() error {
-	// read entire array until ErrEOA (using peekType)
+func (ejvr *extJSONValueReader) skipObject() error {
+	// read entire object until encountering EOA or EOD, and depth is one level below original
 	_, err := ejvr.p.peekType()
+	initialDepth := ejvr.p.depth
 	for err == nil {
 		_, err = ejvr.p.peekType()
+		if err == ErrEOA || err == ErrEOD {
+			err = nil
+			if ejvr.p.depth == initialDepth-1 {
```

Comment:
Gah, I pointed you in the wrong direction. Looking at `advanceState`, `depth` only applies to documents, not arrays. Sorry about that!

Reply:
Ah, you're right, all good!